sklearn/metrics/cluster/supervised.py

```diff
@@ -602,6 +602,8 @@ def mutual_info_score(labels_true, labels_pred, contingency=None):
     contingency_nm = nz_val / contingency_sum
     # Don't need to calculate the full outer product, just for non-zeroes
     outer = pi.take(nzx) * pj.take(nzy)
```

Comment:
Is it worthwhile always casting to int64? Usually there should either be a small contingency matrix, or a sparse one, so I don't think memory is a big issue...

Reply:
IMO, I don't think that's required. But happy to do that if there's some special advantage over the current proposed method.

Reply:
It merely avoids duplicated code

Reply:
@jnothman do you mean casting the `contigency` variable to `int64`? IIRC this is what I was thinking as well.

Reply:
That's certainly an option if contingency is sparse. I just meant removing the if line

Reply:
Do you want the `outer` variable to be always converted to int64?
```Python
outer = pi.take(nzx).astype(np.int64) * pi.take(nzy).astype(np.int64)
```

Reply:
Yes, that's my suggestion